### PR TITLE
rockchip: add support for Radxa ROCK Pi S

### PR DESCRIPTION
**Pull request by RadxaNaoki**
Radxa ROCK Pi S is a small in size, full in features SBC[1] using the
Rockchip RK3308B SoC.

Hardware
--------
- Rockchip RK3308B SoC
- Quad A35 CPU
- 256/512MB DDR3 RAM
- Optional 4/8GB eMMC
- Micro SD Card slot
- Optional WiFi 4 and BT 4 (not supported yet)
- 1x 100M Ethernet with PoE support (additional PoE HAT required)
- 1x USB 2.0 Type-A port (Host)
- 1x USB 2.0 Type-C port (OTG)
- 2x 26 Pin GPIO header

[1] https://radxa.com/products/rockpi/pis

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card or
internal eMMC using dd.

### CODE REVIEW

**Comment by musashino205** [comment]
Please describe installation instruction in the commit message.

https://openwrt.org/submitting-patches#submission_guidelines

> - commit description
>   - If you add support for new hardware: Include in your commit message a short description of the hardware and how to install OpenWrt on it.

**Comment by RadxaNaoki** [comment]
> Please describe installation instruction in the commit message.
> 
> https://openwrt.org/submitting-patches#submission_guidelines
> 
> > * commit description
> >   
> >   * If you add support for new hardware: Include in your commit message a short description of the hardware and how to install OpenWrt on it.

thanks. done.

**Comment by RadxaNaoki** [comment]
now I'm trying to support WiFi/BT (RTL8723DS). please give me a sec...

File: target/linux/rockchip/patches-6.6/900-radxa-rock-pi-s-dts.patch
Comment (hauke) [inline target/linux/rockchip/patches-6.6/900-radxa-rock-pi-s-dts.patch:1]: Please send this patch also to the upstream Linux kernel.

Comment (hauke) [reply target/linux/rockchip/patches-6.6/900-radxa-rock-pi-s-dts.patch:1]: It looks like most of these changes are already integaretd in upstream Linux.

Please just backport the upstream Linux patches:

https://github.com/torvalds/linux/commit/c45de75d7a9a arm64: dts: rockchip: add gpio-line-names to rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/085021cc825e arm64: dts: rockchip: rk3308-rock-pi-s gpio-line-names cleanup
https://github.com/torvalds/linux/commit/1fb98c855ccd arm64: dts: rockchip: Fix SD NAND and eMMC init on rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/d2a52f678883 arm64: dts: rockchip: Rename LED related pinctrl nodes on rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/fc0daeccc384 arm64: dts: rockchip: Add sdmmc related properties on rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/7affb86ef625 arm64: dts: rockchip: Add pinctrl for UART0 to rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/4b64ed510ed9 arm64: dts: rockchip: Add mdio and ethernet-phy nodes to rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/100b3bdee603 arm64: dts: rockchip: Add io-domains to rk3308-rock-pi-s
https://github.com/torvalds/linux/commit/12c3ec878cbe arm64: dts: rockchip: Update WIFi/BT related nodes on rk3308-rock-pi-s

**Comment by RadxaNaoki** [comment]
I decided to drop wi-fi/bt support for now. I'll make new patch.

File: target/linux/rockchip/patches-6.6/900-radxa-rock-pi-s-dts.patch
Comment (RadxaNaoki) [reply target/linux/rockchip/patches-6.6/900-radxa-rock-pi-s-dts.patch:1]: 1st, 2nd, 5th, 6th, and 7th are cherry-picked. 3rd and 4th are already backported to linux-6.6.y.
8th, and 9th are for Wi-Fi/BT, not required for this PR.

**Comment by RadxaNaoki** [comment]
> I decided to drop wi-fi/bt support for now. I'll make new patch.

done. please review for merge.

**Comment by hauke** [comment]
Thank you, I merged the PR inbbcd5111cb8b24ae735085a9530afb9551f34d96. Sorry I accidentally closed it before

**Pull request closed without merging**